### PR TITLE
Implement attach options

### LINKED ISSUE #3477: Silent failure in Postgres extension when using unsupported type
I'm noticing silent failure of the Postgres table scan when using an unsupported type (Kùzu doesn't tell me there's an issue, but the table isn't listed as available even though it exists in Postgres)

I have a Postgres schema in which I want to use the `DECIMAL` data type to represent money values correctly with exactly two decimals, and with 6 significant digits. Here's my desired schema:

```sql
CREATE TABLE IF NOT EXISTS wines(
    wine_id INT PRIMARY KEY,
    title VARCHAR(1028) NOT NULL,
    variety VARCHAR(255) NOT NULL,
    points SMALLINT,
    price DECIMAL(6, 2)
)
```

When I load this data into Postgres, I verify that the table exists, and my SQL query to list the data works.

However, when I list the table in Kùzu's Postgres extension via `CALL show_tables() RETURN *`, the table doesn't appear (it's not accessible via the schema cache, presumably because Kùzu doesn't correctly cast the `DECIMAL` type in Postgres to our internal representation of `DOUBLE`).

I know this is the case, because when I replace the `DECIMAL(6,2)` with a `FLOAT8`, i.e., the following schema, it works:

```sql
CREATE TABLE IF NOT EXISTS wines(
    wine_id INT PRIMARY KEY,
    title VARCHAR(1028) NOT NULL,
    variety VARCHAR(255) NOT NULL,
    points SMALLINT,
    price FLOAT8
)
```

Then, the table appears in Kùzu as follows:

```
┌────────────────┬──────────┬────────────────┬─────────┐
│ name           ┆ type     ┆ database name  ┆ comment │
│ ---            ┆ ---      ┆ ---            ┆ ---     │
│ str            ┆ str      ┆ str            ┆ str     │
╞════════════════╪══════════╪════════════════╪═════════╡
│ wines          ┆ ATTACHED ┆ wine(POSTGRES) ┆         │
└────────────────┴──────────┴────────────────┴─────────┘
```

## Desirable behaviour

If, for some reason, the table that exists in Postgres can't be scanned in Kùzu, we should display an appropriate error so that the user knows something is wrong. In this case, it's not clear why switching the price schema from `DECIMAL(6,2)` to `FLOAT8` actually worked, so that's something to test out.

In the longer run, we should extend support for specialized Postgres data types (like `NUMERIC` or `DECIMAL`) as handling money values in Postgres should never be done with `FLOAT4` or `FLOAT8` as it's fundamentally imprecise due to floating point precision limitations.
https://stackoverflow.com/questions/15726535/which-datatype-should-be-used-for-currency

### PR DESCRIPTION
**Pull request by acquamarin**
This PR implements the attach options framework and adds an attach option: SKIP_INVALID_TABLE.
The option is set to false by default.
If the option is true:
The duckdb/postgres scanner skips tables with unsupported datatypes.
If the option is false:
The duckdb/postgres scanner errors on tables with unsupported types.

Closes #3477

### CODE REVIEW

File: src/antlr4/Cypher.g4 [hunk 1]
```diff
@@ -66,7 +66,13 @@ kU_ImportDatabase
     : IMPORT SP DATABASE SP StringLiteral;
 
 kU_AttachDatabase
-    : ATTACH SP StringLiteral (SP AS SP oC_SchemaName SP)?  (SP? '(' SP? DBTYPE SP StringLiteral SP? ')')?;
+    : ATTACH SP StringLiteral (SP AS SP oC_SchemaName)? SP '(' SP? DBTYPE SP oC_SymbolicName (SP? ',' SP? kU_AttachOptions)? SP? ')' ;
+
+kU_AttachOption
```
Comment (andyfengHKU) [inline src/antlr4/Cypher.g4:71]: Let's reuse kU_ParsingOption(s) and also rename it to KU_Option(s)

File: src/include/parser/parsed_data/attach_info.h [hunk 2]
```diff
@@ -2,11 +2,23 @@
 
 #include <string>
 
+#include "common/case_insensitive_map.h"
+#include "common/types/value/value.h"
+
 namespace kuzu {
 namespace parser {
 
+struct AttachOption {
```
Comment (andyfengHKU) [inline src/include/parser/parsed_data/attach_info.h:11]: Unify with ParsingOption

File: src/parser/transform/transform_attach_database.cpp [hunk 3]
```diff
@@ -1,18 +1,31 @@
...
 #include "parser/transformer.h"
 
 namespace kuzu {
 namespace parser {
 
+AttachOption Transformer::transformAttachOptions(CypherParser::KU_AttachOptionsContext& ctx) {
+    AttachOption option;
+    for (auto& attachOption : ctx.kU_AttachOption()) {
+        auto key = transformSymbolicName(*attachOption->oC_SymbolicName());
+        auto value = transformLiteral(*attachOption->oC_Literal())
```
Comment (andyfengHKU) [inline src/parser/transform/transform_attach_database.cpp:12]: Normally you don't wanna perform any check/cast here. Just pass ParsedExpression and bind it later on

File: extension/duckdb/src/include/duckdb_storage.h [hunk 4]
```diff
@@ -8,7 +8,11 @@ namespace duckdb_extension {
 
 class DuckDBStorageExtension final : public storage::StorageExtension {
 public:
-    static constexpr const char* dbType = "DUCKDB";
+    static constexpr const char* DB_TYPE = "DUCKDB";
+
+    static constexpr const char* SKIP_UNSUPPORTED_TABLE_KEY = "SKIP_UNSUPPORTED_TABLE";
```
Comment (andyfengHKU) [inline extension/duckdb/src/include/duckdb_storage.h:13]: Up to u. Do we want to maintain all extension config into its own class?

**Review by andyfengHKU — APPROVED** [review]

File: extension/duckdb/src/include/duckdb_storage.h (hunk 4 above)
Comment (acquamarin) [reply extension/duckdb/src/include/duckdb_storage.h:13]: Yes, these constants are only be used by duckdb extensions.

**Pull request merged**